include/hyprutils/animation/AnimatedVariable.hpp

```diff
@@ -0,0 +1,281 @@
+#pragma once
+
+#include <functional>
+#include <chrono>
+
+namespace Hyprutils {
+    namespace Animation {
+        class CAnimationManager;
+
+        /*
+            Structure for animation properties.
+            Config properties need to have a static lifetime to allow for config reload.
+        */
+        struct SAnimationPropertyConfig {
+            bool                      overridden = true;
+
+            std::string               internalBezier  = "";
+            std::string               internalStyle   = "";
+            float                     internalSpeed   = 0.f;
+            int                       internalEnabled = -1;
+
+            SAnimationPropertyConfig* pValues          = nullptr;
```

Comment:
While we are rewriting this, can these be made into WP for memory safety?

Reply:
that's fine, I just want to get rid of raw pointers in the codebase as much as we can